core/src/main/java/nl/inl/util/LuceneUtil.java

```diff
@@ -327,15 +327,13 @@ public static void getFrequenciesFromTermVector(IndexReader reader, int doc,
                                 for (BytesRef term = termsEnum.term(); term != null; term = termsEnum.next()) {
                                     if (maxResults < 0 || results.size() < maxResults) {
                                             String termText = term.utf8ToString();
-                                            String optDesensitized = termText;
-                                            if (!sensitive)
-                                                    optDesensitized = StringUtil.stripAccents(termText).toLowerCase();
-                                            if (!allTerms && !optDesensitized.substring(0, prefix.length()).equalsIgnoreCase(prefix)) {
+                                            boolean startsWithPrefix = sensitive ? termText.startsWith(prefix): StringUtil.insensitiveStartsWith(termText, prefix);
```

Comment:
'prefix' has already been desensitized, but is desensitized again each time this method is executed. Just seems a bit wasteful to me. Why not have a StringUtil.desensitize(), only desensitize termText and compare it to the already-desensitized prefix?

(Of course, we could get into when you should or shouldn't optimize code, but in my opinion there's a difference between premature optimization and writing the best-performing code you can from the start, without making it convoluted of course)

Reply:
sorry, overlooked ```StringUtil.stripAccents(prefix).toLowerCase();```

I'll adapt